src/runtime/converter.cs

```diff
@@ -133,6 +134,22 @@ internal static IntPtr ToPython(object value, Type type)
                 return result;
             }
 
+            if (value is IList && value.GetType().IsGenericType)
+            {
+                using (var resultlist = new PyList())
+                {
+                    foreach (object o in (IEnumerable)value)
```

Comment:
Agreed, I opened a separate issue for this dead code: https://github.com/pythonnet/pythonnet/issues/428